Add Ruff check GitHub Actions push and pull request merge block

EdwardTheLegend opened the pull request:
Added the `ruff.yml` for the GitHub Actions workflow and also had to force fixes to solve linter violations such as removing unused imports.

Closes #35

EdwardTheLegend committed:
Add ruff github actions workflow

EdwardTheLegend committed:
ruff format and remove all unused imports

EdwardTheLegend committed:
Set ruff line-length to 95

ilikecubesnstuff commented on bot.py:
```diff
@@ -6,46 +6,52 @@
+    "dev",
+    "embeds",
+    "filter",
     # 'help',
```
I assume Ruff missed this because it's commented out, but change these quotation marks as well.

EdwardTheLegend committed:
run formatting with new line length 95

NathanealV commented on cogs/monitor.py:
```diff
@@ -183,34 +213,42 @@ async def message_delete(self, ctx: commands.Context, pattern_id: int):
+            f"{'Disabled' if monitored_message.disabled else 'Enabled'} pattern with ID {monitored_message.monitor_message_id} in monitor."
+        )
+        await ctx.send(
+            f"The pattern (`{monitored_message.message}`) [ID: {monitored_message.monitor_message_id}] has been successfully {'disabled' if monitored_message.disabled else 'enabled'}."
```
Line 241 + line 305 are over the 95 character line length (may involve other lines as well). Should probably manually adjust these.

NathanealV reviewed (commented):
Thank you for the PR! Changes look good - might need to go over some files again with f-strings as they might be over the line length (mostly with `cogs/monitor.py`.

EdwardTheLegend commented:
I ran the formatting with the new line length on the whole repo. I think the exceptions you mentioned might be where the ruff rules based on pep standards don't recommend folding on. Some things like long strings aren't usually broken over multiple lines even if they go a bit over the limit.

ilikecubesnstuff reviewed (changes requested):
Apart from my comment and Nath's, this looks fine now. I'll approve after the changes are pushed. Thanks!

ilikecubesnstuff committed:
Pull request changes:
- Change quotations in commented code.
- Reformat long strings to obey line limit.
Additionally, update package versions in lock file.

NathanealV reviewed (approved):
LGTM!
